openghg/store/_flux.py

```diff
@@ -228,20 +212,23 @@ def read_file(
         if optional_metadata is None:
             optional_metadata = {}
 
-        # Make sure none of these are Nones
-        to_add = {k: v for k, v in optional_keywords.items() if v is not None}
-
-        # warn if `optional_metadata` overlaps with keyword arguments
-        overlap = [k for k in optional_metadata if k in to_add]
-        if overlap:
-            msg = (
-                f"Values for {', '.join(overlap)} in `optional_metadata` are "
-                "being overwritten by values passed as keyword arguments."
-            )
-            logger.warning(msg)
+        # TODO: REMOVE ONCE OPTIONAL KEYS CAN BE TAKEN FROM OTHER INPUTS THAN OPTIONAL_METADATA
```

Comment:
I think the original intention was to use `optional_metadata` for optional keys that were added to the default config, and the optional keys that are in the default config would remain in the function signatures.

Allowing optional keys to be passed by other means makes sense (at least from `additional_input_parameters`)